docs/performance-testing.md

```diff
@@ -0,0 +1,89 @@
+# Performance Testing
+
+TEAMMATES makes use of [JMeter](https://jmeter.apache.org/) for performance testing.
+The relevant files are located in the `src/jmeter/` directory. The JMeter tests are the `.jmx` files located in the `src/jmeter/tests/` directory.
+
+TEAMMATES uses the [`jmeter-gradle-plugin`](https://github.com/jmeter-gradle-plugin/jmeter-gradle-plugin) for automating the processes of running performance tests and generating reports.
+
+## Performance Test Workflow
+
+Before running a performance test, you need to setup the relevant data in the GAE datastore. 
+To facilitate this, there are Java scripts in the `teammates.performance.scripts` package (`src/jmeter/java/teammates/performance/scripts/` directory).
+
+There are 3 steps involved in doing a performance test:
+
+1. Setting up the data: This is done by executing the relevant Java script in the `setup` package (e.g. `SetupStudentProfileTest`).
```

Comment:
can you run the script on `Windows`? Cos I tried running the test file on Mac and it throws an error when creating the data directory

Reply:
More importantly: can you run the script in bash/shell?

Reply:
> can you run the script on Windows?

It works for me when I run through IntelliJ.
Can you debug and find out why it's not working for you? The data folder is supposed to be created through lnp/util/Test.Properties#createTestDataFolder

Reply:
> More importantly: can you run the script in bash/shell?

@wkurniawan07 I don't think we can, unless everything is packaged into a jar.

Even if we go to `out/test/classes/teammates/...` and do `javac`, it will not work due to dependencies on external libraries

Reply:
> More importantly: can you run the script in bash/shell?

@RonakLakhotia Can you look at this? We'll have to create a gradle task